### DISCUSSION TITLE
Cannot connect to Webserver

### DISCUSSION THREAD

**Discussion by achronoxx** [PlummersSoftwareLLC/NightDriverStrip#585] [Q&A] [answered]
Using web installer, I just loaded NightDriverStrip (Ledstrip(N,S)) to my ESP32 (WEMOS LoLin32 V1.0.0) and I cannot connect it to the webserver when I enter the ESP32 IP in a browser window. 

I know the ESP32 is connected to Wifi, as the log windows shows the following: 
(I) (loop)(C1) WiFi: WL_CONNECTED, IP: 192.168.0.128, Mem: 63760, LargestBlk: 36852, PSRAM Free: 0/0, LED FPS: 22 LED Bright: 100%, LED Watts: 30, Buffer: 0/22, CPU: 001%, 002%, FreeDraw: 0.008

I also tested the ESP32 with another app (from Arduino IDE) and confirmed it connects an communicates over WiFi.

Please let me know if I'm doing something wrong?

Also, I tried loading the "Demo" project and it loaded successfully, but I could not ever get the screen to configure WiFi.  As suggested,  I  tried going in and out of the Logs/Console window multiple times with no luck.

Thanks,
Andrew

**Comment by rbergen** [comment]
In the web installer, the letters behind the project name indicate which key features are enabled in the project you're flashing. "N,S" means Networking and Socket Server are enabled, but the webserver isn't (which would be indicated by a "W" being included in the list). In a similar vein, the absence of any feature letters in the Demo project means that networking isn't included at all - which is why the ability to set up WiFi isn't offered.

**Reply by achronoxx** [reply]
Hi Rutger, thanks for the help!

I've now loaded the "TTGO (N,W,S,C,R)" project on the same ESP32 (WEMOS Lolin32) and I was able to connect to the webserver, however, there are a number of issues:
1. The Webserver was active, but only one effect (Spectrum Fade) was active and I could not see a way to add other effects.  I also could not change any of the parameters for the active (Spectrum Fade) effect.
2. I connected the ESP32 to my WS2812B strip (150 pixels), and all the LEDs are dark.
3. I'm sure I have the correct output (GPIO 21, as per globals.h) from the ESP32, as I used my oscilloscope to confirm it is the expected WS2812 signal, and I do see a ~23ms burst of data and all bits are logic-0.  
4. Other than the first time, I can no longer connect to the Webserver, and I've double-checked that I have the correct IP address.  It also fails to connect when I use the web-installer option to "Visit Device".
Note that the LOGS&Console do show active data, and I can also connect to ESP32 with a WiFi Telnet Terminal to look at debug data, so I know the WiFi is working.

Any idea, how I can re-establish a connection to Webserver, and how I can change/add effects once I connect?  I tried resetting and power-cycling ESP and it did not help.  If I erase and reload the project, it will reconnect to server, but this is not a good solution.

Thanks,
Andrew

**Reply by rbergen** [reply]
Hi Andrew, you're welcome.

1. The effects that are available for a project depend on the project - this is determined by the contents of effects.cpp specifically. For the TTGO project you flashed, the only effect built into the image is indeed Spectrum Fade.
2. You seem to be flashing a project that's intended for the LilyGO TTGO on a WEMOS Lolin32. I'm pretty sure no one I know as being involved with this project has ever tried that combination before. Which doesn't mean it can't work, but it does mean no one has _seen_ it work. For example, for audio-driven effects to light up not only the output pin needs to be configured correctly, but the whole microphone setup needs to be as well. That's one of the things that is known to be device-specific.
3. See 2.
4. There's no way for me to know why your web server stopped responding without at least seeing the logging your device generates. It's not unusual for things like these to be related to low memory conditions -&nbsp;we've seen some of that in this project in recent times&nbsp;-, but it could be something else entirely as well.

**Reply by rbergen** [reply]
Quickly circling back to this, I just noticed that the TTGO project is listed under the ESP32 device in the web installer. That makes the confusion understandable. I'll be moving it to a separate, new LilyGO TTGO device.

**Comment by robertlipe** [comment]
Please include a device name.

(Liligo is a brand, Like Adafruit.)

Last. I looked, they had a couple of models called T-Display, like one with
an S3 and one with a touch screen and another with a super resolution and
one with a coffe maker or something. Are they compatible enough for our
purposes? Without hardware in hand to try or a reason to stare at
schematics, I won't guess.

Since. We suspect that a. Change in mesmerizer effect didn't change a strip
build (indeed, as in the other thread,.we have reason to wonder if it made
a difference in Mesmerizer...) Should we run a pio pkg (or whatever it's
called) and get the working versions and enforce those minimum version
numbers in piotrans.ini just to narrow the mix of combinations of things
we're all supporting?

If we're depending on ***@***.***
And a leak was. Introduced in 1.2.6 two weeks ago but fixed in 1.2.7 this
morning (all packages and numbers pulled from a hat) maybe we just raise
the minimum everywhere to the version you just qualified. Without (really)
expensive debugging I'm more likely to blame external packages (freakin'.
Arduino 😉) than a Mesmerizer effect.

If you pre-approve the idea but don't want to put fingers on keyboard, I
can do that later today and land a toasty fresh PR into the queue for you
to throw into your next batch.

RJL

**Reply by heidepiek** [reply]
@robertlipe I think this is the good version of the ttgo from lilygo.  (LilyGO TTGO T-Display V1.1 ESP32 - with 1.14 inch TFT Display)
 Lilygo has several variants, but I think dave used this version for the spectrum analyzer. I also own a lilygo T-DISPLAY S3 and that version has a bigger screen.
![lilygo1](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/f0470dd6-8a36-47f8-bb41-4dd78e624b2a)
![lilygo2](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/bf4349d9-9776-44f8-80c1-1ca3a2c5025d)
![lilygo3](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/5ece8dd5-92d8-4698-ac4c-c9d05c0fbde5)
![lilygo4](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/82bb0a10-151d-4dcb-a2d2-96856ec75a41)

**Reply by rbergen** [reply]
I stuck the TTGO project under a device I've called "LilyGO TTGO" for now. I know that still covers a few of their devices, but I honestly don't know exactly which TTGO @davepl used. At least it kind of excludes everything else. If and when I get confirmation concerning (an) exact device(s), I can easily update the device name to something more specific also.

**Reply by achronoxx** [reply, marked as answer]
Hi Rutger,

Thanks for clarifying.  So it seems I need to build my own custom version of the SW if I want to try all the different effects, or is there a plan for someone to add a more general purpose project on the web installer that works with a generic ESP32?.

I was thinking (hoping) that nightdriver would be more similar to WLED app that allows you to pick and choose all the effects from the user GUI, but I'm OK with making my own project with platformIO.

I will install platformIO and try this.

Thanks for the help.

Andrew
